### PR TITLE
(chores) camel-core: setup and cleanup the data after test execution

### PR DESCRIPTION
**Pull request by orpiske**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: core/camel-core/src/test/java/org/apache/camel/component/file/FileProducerCharsetUTFtoISOConvertBodyToTest.java [hunk 1]
```diff
@@ -35,12 +39,27 @@ class FileProducerCharsetUTFtoISOConvertBodyToTest extends ContextTestSupport {
...
+    @AfterEach
+    void cleanupFile() {
+        try {
+            Files.delete(testFile("output.txt"));
+        } catch (IOException e) {
+            fail("The test cannot run due to an error cleaning up: " + e.getMessage());
+        }
+    }
+
+    @RepeatedTest(10)
```
Comment (essobedo) [inline core/camel-core/src/test/java/org/apache/camel/component/file/FileProducerCharsetUTFtoISOConvertBodyToTest.java:61]: Is it related?

**Review by essobedo — COMMENTED** [review]
You could consider using the TempDirectory extension https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory

**Comment by orpiske** [comment]
> You could consider using the TempDirectory extension https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory

Yeah, but at later moment. I think there's already some code using it (or, at least, a ticket about this).

File: core/camel-core/src/test/java/org/apache/camel/component/file/FileProducerCharsetUTFtoISOConvertBodyToTest.java (hunk 1 above)
Comment (orpiske) [reply core/camel-core/src/test/java/org/apache/camel/component/file/FileProducerCharsetUTFtoISOConvertBodyToTest.java:61]: Yes it is. This test was flaky on some environment. I want to make sure it runs well after this change.

**Review by essobedo — APPROVED** [review]

**Pull request merged**